File: recursion/python/reverse_string.py
# reverse values in a string

def reverse_string_iterative(string):
    reversed = ''

    for c in string:
        reversed = c + reversed

    return reversed


def reverse_string_recursive(string):
    if len(string) == 0 or len(string) == 1:
        return string
    else:
        head = string[0]
        tail = string[1:]

        return reverse_string_recursive(tail) + head


if __name__ == '__main__':
    print(reverse_string_iterative('hello'))
    print(reverse_string_recursive('hello'))

File: recursion/javascript/reverseString.js
// reverse the order of a string

function reverseStringIterative(string) {
    let reversed = '';

    for (char of string) {
        reversed = char + reversed;
    }

    return reversed;
}

function reverseStringRecursive(string) {
    if (string.length === 0 || string.length === 1) {
        return string;
    } else {
        const head = string[0];
        const tail = string.slice(1, string.length);

        return reverseStringRecursive(tail) + head;
    }
}

console.log(reverseStringIterative('hello'));
console.log(reverseStringRecursive('hello'));
